src/flask/config.py

```diff
@@ -332,15 +332,14 @@ def get_namespace(
         """
         rv = {}
         for k, v in self.items():
-            if not k.startswith(namespace):
-                continue
-            if trim_namespace:
-                key = k[len(namespace) :]
-            else:
-                key = k
-            if lowercase:
-                key = key.lower()
-            rv[key] = v
+            if k.startswith(namespace):
+                if trim_namespace:
+                    key = k[len(namespace) :]
```

Comment:
Can be done through the `lstrip` command. The implementation will be:

`key = k.lstrip(namespace)`